starfish/test/image/test_imagestack_index.py

```diff
@@ -0,0 +1,134 @@
+from collections import OrderedDict
+
+import numpy as np
+
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Coordinates, Indices
+
+
+def test_indexing_by_r_ch_z():
```

Comment:
Most of your testing can center around `calc_new_physical_coords` to make the tests simpler and easier to comprehend.

Reply:
I'll still need to create Imagestacks with coordinates if I'm just testing calc_new_physical_coords,  which I think is what takes up a lot of the complexity...the test flow is just 1.) Create an imagestack 2.) index it 3.) verify that the new shape is what we expect. And then for  test_indexing_by_x_y 1.) create and image stack with simple coords 2.) index it 3.)verify that the new shape and coords are what we were expecting. I tried to add a lot of comments to explain what was going on, but are you saying I shouldn't test the full indexing of an imagestsck at all? and just create an imagestack with coords and verify that calc_new_coords is working?

Reply:
I think there's value in making `calc_new_physical_coords` easily testable (and indeed build tests for that).  And there's value in testing the `sel` functionality.  Both are very crucial components that deserve some tests.